### PR TITLE
Distributed File Systems, Search Engine Original Papers

### PR DESCRIPTION
**Pull request by ramlaxman**
## Paper Title:
New Folders of: 
- Distributed File Systems
- Search Engine Original Papers

### Paper Year:
Various

### Reasons for including paper
- To remove ambiguity from readers concept.
- Help to clear concepts.

### CODE REVIEW

**Comment by NewAlexandria** [comment]
The Java, C, C++, and Go files are missing links to the papers in your PR.

**Comment by ramlaxman** [comment]
Oh ! I actually added today morning after submitting PR. Also made updation now. Kindly check and let me know.

> 
> 
> The Java, C, C++, and Go files are missing links to the papers in your PR.
> 
> Other than that, I like this PR

**Comment by DarrenN** [comment]
@ramlaxman we already have the Bitcoin paper here: https://github.com/papers-we-love/papers-we-love/blob/7c801a1090fc8ae75a8fc35ebc2881001f9fa436/digital_currency/README.md I would suggest updating that README with better titles / descriptions rather than creating a new folder.

**Comment by ramlaxman** [comment]
Made changes as per your suggestions. Kindly take a look.

File: distributed-file-systems/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Distributed File Systems
+
+- [Sun NFS](https://github.com/ramlaxman/papers-we-love/blob/master/distributed-file-systems/sandbergnfs.pdf)
```
Comment (deltamualpha) [inline distributed-file-systems/README.md]: Explicitly linking to your fork here is wrong; use relative links, please.

**Comment by hakutsuru** [comment]
This is close to usable. I expect to be merged with minimal modifications soon.

Please remove the PDFs included. While we could probably include some of the PDFs, associated READMEs would need to be updated to reference the files. Most importantly, we would need to ensure every PDF could be legally included in the repo, which may require significant review.

Also, please change "[Java Whitepaper from Strastroup website]" to "[Java Whitepaper]" (since we may change the source link someday).

Since you put a lot of work into this, I could make these final changes for you. Thanks for your perseverance on this contribution.

**Comment by ramlaxman** [comment]
Hi @hakutsuru,
Thanks for the review. I will try to avoid these mistakes in future PRs.

**Comment by hakutsuru** [comment]
Thanks, @ramlaxman 

I am merging and fixing this.

****

This should not be merged in this state. But rather than lose this work, I will attempt cleanup.

*Relative link* mean that if you include the PDF, the link is to the hosted paper...
```
* :scroll: [Fast Error-bounded Lossy HPC Data Compression with SZ](fast_error_bounded_Lossy_hpc_data_compression_with_sz.pdf)
```

External links are valuable, but we tend to exclude those when hosting the paper. When hosting a paper, we must make sure that is allowed by copyright. This requires care. While hosting/vetting these papers would be worthwhile, it would be laborous, and since we have links, using those is pragmatic.

This is several pull requests mixed together. Smaller pull requests which are thematically consistent are easier to fix and complete. Many valuable resources added here are not mentioned at all in the pull request description.

**Pull request merged**